torch/utils/benchmark/utils/timer.py

```diff
@@ -135,16 +201,32 @@ def __init__(
         description: Optional[str] = None,
         env: Optional[str] = None,
         num_threads: int = 1,
+        language: str = PYTHON,
     ):
         if not isinstance(stmt, str):
             raise ValueError("Currently only a `str` stmt is supported.")
 
-        # We copy `globals` to prevent mutations from leaking, (for instance,
-        # `eval` adds the `__builtins__` key) and include `torch` if not
-        # specified as a convenience feature.
         globals = dict(globals or {})
-        globals.setdefault("torch", torch)
-        self._globals = globals
+        if language.lower() == PYTHON.lower():
```

Comment:
This wasn't how I expected you were going to fix it. This is code is especially strange to me because PyTorch *doesn't* accept the capitalized variants of device names:

```
>>> torch.zeros(0, device='CPU')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Expected one of cpu, cuda, mkldnn, opengl, opencl, ideep, hip, msnpu, xla device type at start of device string: CPU
```

Reply:
Also, if you're going to go through the trouble of converting an internal rep, might as well define an actual honest to goodness Enum for it; it's more type safe.